src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -76,7 +80,7 @@ public async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(Cancellati
                         return GetDiagnosticData();
                     }
 
-                    var documentIds = DocumentId != null ? [DocumentId] : project.DocumentIds;
```

Comment:
@mavasani here is where we hardcoded that if you didn't pass in a document, you got *only* the diagnostics for the normal Documents (not including additional documents).